src/Engine/GameEvents/MassDamageEvent.java

```diff
@@ -0,0 +1,82 @@
+package Engine.GameEvents;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import Engine.XYCoord;
+import Terrain.MapMaster;
+import UI.MapView;
+import UI.Art.Animation.GameAnimation;
+import Units.Unit;
+
+/**
+ * Deals damage of configurable lethality to an arbitrary number of units, without invoking combat
+ * Note: Damage dealt will not be tracked correctly if the event is executed twice
+ */
+public class MassDamageEvent implements GameEvent
+{
+  // Records how many HP each victim lost; doubles as our victim storage area
+  private Map<Unit, Integer> victims = new HashMap<Unit, Integer>();
+  public final int damage;
+  public final boolean lethal;
+
+  public MassDamageEvent(Collection<Unit> pVictims, int pDamage, boolean isLethal)
+  {
+    for(Unit victim : pVictims)
+    {
+      victims.put(victim, 0);
+    }
+    damage = pDamage;
+    lethal = isLethal;
+  }
+
+  @Override
+  public GameAnimation getEventAnimation(MapView mapView)
+  {
+    return null;
+  }
+
+  @Override
+  public void sendToListener(GameEventListener listener)
+  {
+    listener.receiveMassDamageEvent(victims);
+  }
+
+  @Override
+  public void performEvent(MapMaster gameMap)
+  {
+    for (Unit victim : victims.keySet())
+    {
+      int starting = victim.getHP();
+      if( lethal && damage > starting )
+      {
+        victim.damageHP(-damage);
+        victims.put(victim, starting);
+
+        // Guess he's not gonna make it.
+        // TODO: Is there a better way to do this?
```

Comment:
This kind of logic (determining whether to create a given event) should probably be done at the `GameAction.getEvents()` level, not on the fly within a different event's execution.

It seems like it should be pretty straightforward to loop over the victims in `ExplodeAction` to see which ones will be killed, and adding `UnitDieEvent` for each. Maybe only use the `MassDamageEvent` for the ones that don't. I'd consider calling it just `UnitDamageEvent` for consistency and generality.

Reply:
This was made under the (highly mistaken) impression that `GameAction.getEvents()` uses a generic `GameMap` rather than a `MapMaster`. I'd suggest we make this easier to remember somehow, but I don't think you can fix stupid.

It would be straightforward, for sure.
I think the `MassDamageEvent` should still apply to the killed units, since the point of publishing the event is to notify listeners of who was affected by the damage.

...I guess I can't argue too hard with `UnitDamageEvent`, but I put the "Mass" in there to specify that it's affecting more than one unit at a time.

Reply:
...On that note, I guess I ought to add [defeat] to both the damage event and the delete event, since they can both kill the last surviving unit.

Reply:
I could see how `MassDamageEvent` might be more clear in the CO event listener, since I think battle-damage charge is normally handled elsewhere. Maybe we can consolidate those and reduce ambiguity even more?

Reply:
I don't think we can consolidate these. The charge for combat takes into account both funds damage dealt and taken (with different multipliers) and the amount of HP removed from the enemy you attacked.
I guess you could just use the same listening function but with only the "attacked" unit filled out, but that would break a lot of assumptions and be really ugly IMO.

Reply:
Yeah, I agree that `MassDamageEvent` should apply to killed units.

Which you call the event is up to you, I think either works well.

Yes, mass damage/delete should both have game-ending potential.

Yeah, I guess we don't have to consolidate the damage stuff.